src/seedu/addressbook/Main.java

```diff
@@ -107,7 +114,9 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
-            storage.save(addressBook);
+            if (!storage.save(addressBook)){
```

Comment:
there shouldn't be a need to check the storage here. it should be as simple as a class method call. Let the catch block handle the exceptions